### PR TITLE
[phee-605] make Reedem and pay work with ExternalLookup flag

### PR DESCRIPTION
**Pull request by Anover000**
[PHEE-605 make Reedem and pay work with ExternalLookup flag](https://fynarfin.atlassian.net/browse/PHEE-605?atlOrigin=eyJpIjoiM2NkNDE1YjNjMGExNGM3ZWEwYmUwMmFkMjUxNDI0N2IiLCJwIjoiamlyYS1zbGFjay1pbnQifQ)

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!
- [x] Followed the PR title naming convention mentioned above.

- [ ] Design related bullet points or design document link related to this PR added in the description above.

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [ ] Created/updated unit or integration tests for verifying the changes made.

- [ ] Added required Swagger annotation and update API documentation with details of any API changes if applicable

- [ ] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

### CODE REVIEW

File: src/main/java/org/mifos/pheevouchermanagementsystem/data/VoucherValidator.java [hunk 1]
```diff
@@ -164,11 +164,11 @@ public PhErrorDTO validateRedeemVoucher(RedeemVoucherRequestDTO request) {
                 .isNullWithFailureCode(VoucherValidatorsEnum.INVALID_AGENT_ID)
                 .validateFieldMaxLengthWithFailureCodeAndErrorParams(expectedAgentIDLength, VoucherValidatorsEnum.INVALID_AGENT_ID_LENGTH);
 
-//        // Check for voucherSecretNumber
-//        validatorBuilder.reset().resource(resource).parameter(voucherSecretNumber).value(request.getVoucherSecretNumber())
-//                .isNullWithFailureCode(VoucherValidatorsEnum.INVALID_VOUCHER_SECRET_NUMBER)
-//                .validateFieldNotBlankAndLengthWithFailureCodeAndErrorParams(expectedVoucherSecretNumberLength,
-//                        VoucherValidatorsEnum.INVALID_VOUCHER_SECRET_NUMBER_LENGTH);
+        // // Check for voucherSecretNumber
```
Comment (fynmanoj) [inline src/main/java/org/mifos/pheevouchermanagementsystem/data/VoucherValidator.java:167]: delete these lines if not needed

File: src/main/java/org/mifos/pheevouchermanagementsystem/camel/routes/AccountLookupRoute.java [hunk 2]
```diff
@@ -29,6 +30,7 @@ public void configure() throws Exception {
             String callbackUrl = exchange.getProperty(CALLBACK, String.class);
             exchange.getIn().setHeader(CALLBACK, callbackUrl);
             exchange.getIn().setHeader("X-Registering-Institution-ID", exchange.getProperty(REGISTERING_INSTITUTION_ID));
+            exchange.getIn().setHeader("isExternalLookup", exchange.getProperty(IS_EXTERNAL_LOOKUP));
```
Comment (fynmanoj) [inline src/main/java/org/mifos/pheevouchermanagementsystem/camel/routes/AccountLookupRoute.java:33]: are we still using the header approach?

File: src/main/java/org/mifos/pheevouchermanagementsystem/data/VoucherValidator.java [hunk 3]
```diff
@@ -128,12 +128,12 @@ public PhErrorDTO validateVoucherLifecycle(RequestDTO request) {
             request.setVoucherInstructions(new ArrayList<>(Arrays.asList(new VoucherInstruction())));
         }
 
-//        request.getVoucherInstructions().forEach(voucherInstruction -> {
-//            // Check for serialNumber
-//            validatorBuilder.reset().resource(resource).parameter(serialNumber).value(voucherInstruction.getSerialNumber())
-//                    .isNullWithFailureCode(VoucherValidatorsEnum.INVALID_SERIAL_NUMBER).validateFieldMaxLengthWithFailureCodeAndErrorParams(
-//                            expectedSerialNumberLength, VoucherValidatorsEnum.INVALID_SERIAL_NUMBER_LENGTH);
-//        });
+        // request.getVoucherInstructions().forEach(voucherInstruction -> {
```
Comment (fynmanoj) [inline src/main/java/org/mifos/pheevouchermanagementsystem/data/VoucherValidator.java:131]: delete these lines if not needed

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/main/java/org/mifos/pheevouchermanagementsystem/camel/routes/AccountLookupRoute.java (hunk 2 above)
Comment (Anover000) [reply src/main/java/org/mifos/pheevouchermanagementsystem/camel/routes/AccountLookupRoute.java]: No, there were some left over parts, which I have removed now.

File: src/main/java/org/mifos/pheevouchermanagementsystem/data/VoucherValidator.java (hunk 3 above)
Comment (Anover000) [reply src/main/java/org/mifos/pheevouchermanagementsystem/data/VoucherValidator.java:131]: these are some validation code which might be use later

**Review by fynmanoj — APPROVED** [review]

**Pull request merged**